Sources/Networking/HTTPClient/HTTPResponse.swift

```diff
@@ -14,48 +14,108 @@
+    var responseHeaders: HTTPClient.ResponseHeaders { get }
+    var body: Body { get }
+    var requestDate: Date? { get }
+
+}
+
+struct HTTPResponse<Body: HTTPResponseBody>: HTTPResponseType {
+
+    var statusCode: HTTPStatusCode
     var responseHeaders: HTTPClient.ResponseHeaders
     var body: Body
     var requestDate: Date?
-    var verificationResult: VerificationResult
 
 }
 
 extension HTTPResponse: CustomStringConvertible {
 
-    var description: String {
-        let body: String = {
-            if let bodyDescription = (self.body as? CustomStringConvertible)?.description {
-                return bodyDescription
-            } else {
-                return "\(type(of: self.body))"
-            }
-        }()
+    fileprivate var bodyDescription: String {
+        if let bodyDescription = (self.body as? CustomStringConvertible)?.description {
+            return bodyDescription
+        } else {
+            return "\(type(of: self.body))"
+        }
+    }
 
+    var description: String {
         return """
         HTTPResponse(
             statusCode: \(self.statusCode.rawValue),
-            body: \(body),
+            body: \(self.bodyDescription),
+            requestDate: \(self.requestDate?.description ?? "<>")
+        )
+        """
+    }
+
+}
+
+// MARK: - VerifiedHTTPResponse
+
+struct VerifiedHTTPResponse<Body: HTTPResponseBody>: HTTPResponseType {
+
+    var response: HTTPResponse<Body>
+    var verificationResult: VerificationResult
+
+    init(response: HTTPResponse<Body>, verificationResult: VerificationResult) {
+        self.response = response
+        self.verificationResult = verificationResult
+    }
+
+    init(
+        statusCode: HTTPStatusCode,
+        responseHeaders: HTTPClient.ResponseHeaders,
+        body: Body,
+        requestDate: Date? = nil,
+        verificationResult: VerificationResult
+    ) {
+        self.init(
+            response: .init(
+                statusCode: statusCode,
+                responseHeaders: responseHeaders,
+                body: body,
+                requestDate: requestDate
+            ),
+            verificationResult: verificationResult
+        )
+    }
+
+}
+
+extension VerifiedHTTPResponse: CustomStringConvertible {
+
+    var description: String {
+        return """
+        VerifiedHTTPResponse(
+            statusCode: \(self.statusCode.rawValue),
+            body: \(self.response.bodyDescription),
+            requestDate: \(self.requestDate?.description ?? "<>")
             verification: \(self.verificationResult)
         )
         """
     }
 
 }
 
-extension HTTPResponse {
+// MARK: - Extensions
+
+extension HTTPResponseType {
 
     /// Equivalent to `HTTPURLResponse.value(forHTTPHeaderField:)`
     /// In keeping with the HTTP RFC, HTTP header field names are case-insensitive.
-    func value(forHeaderField field: String) -> String? {
```

Comment:
A nice refactor here: this is now defined on the protocol `HTTPResponseType` so it's available for both response types.
Also no longer accepting any `String`, enforcing that we use this only with `HTTPClient.ResponseHeader`s and not `HTTPClient.RequestHeader`s